woocommerce/woocommerce-gutenberg-products-block
# Update checkout API endpoint to process orders

This might be more work when we integrate it, but this should be good enough to merge for now to unblock other tasks.

Creates the `wc/store/checkout` endpoint. This endpoint takes an existing order and processes payment using a gateway.

To make this easier to test, add this to wp-config:

```
define( 'WOOCOMMERCE_STORE_API_DISABLE_NONCE_CHECKS', true );
```

This disables nonce verification so you can use basic auth from a REST client to test the endpoints.

After doing this:

1. Enable the bacs gateway
2. Add some things to your cart logged in
3. Using basic auth, POST to `/wc/store/cart/create-order`. Note the order ID and key in the response.
4. Use the order ID and key to POST to `wc/store/checkout`. The request will be something like this:

```json
{
  "order_id": 100,
  "order_key": "xxxxxx",
  "payment_method": "bacs"
}
```

Check the response and confirm it was successful.

Closes #1938
